### DISCUSSION TITLE
What's the relationship between targets and tarchetypes?

### DISCUSSION THREAD

**Discussion by hadley** [ropensci/targets#1319] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

It's totally possible I've missed some docs somewhere, but I'd love to know more about the relationship between these packages. Do you expect most people will use tarchetypes?

**Comment by wlandau** [comment]
I designed the interface of targets with [target factories](https://wlandau.github.io/targetopia/contributing.html#target-factories) in mind (though I originally called them "archetypes") and DSLs. My hope was that third-party packages would contribute target factories to make `targets` easier to adapt to specific use cases. For example, [`stantargets`](https://docs.ropensci.org/stantargets/) helps users create [large simulations with Stan models](https://docs.ropensci.org/stantargets/articles/simulation.html) without needing to understand dynamic branching, [`geotargets`](https://njtierney.github.io/geotargets/) makes `targets` "just work" (TM) with `terra` raster data. 

Static branching, literate programming, etc. naturally sit at the same level: they make `targets` easier to use, and they naturally decouple from the main package. The goal of `tarchetypes` is to be a home for this general-purpose syntactic sugar. I do expect most `targets` users to find it helpful.

**Reply by hadley** [reply]
To me that sounds like the functions in targets are mostly used to make targets work, and the end-user will mostly use functions from tarchetypes?

**Reply by wlandau** [reply]
It varies, but most `_targets.R` files do have a lot of calls to `tarchetypes` functions.

**Reply by Aariq** [reply]
Yeah, it's definitely possible to create a targets pipeline without `tarchetypes`, but there are a few helper functions in `tarchetypes` I almost always use like `tar_plan()` and `tar_file()`

**Reply by hadley** [reply]
In that case, I think you should consider either moving the most important/most commonly used tarchetype functions to targets, or wherever you show examples show loading both targets and tarchetypes. Otherwise I think it's an unnecessary burden to the new user.

**Reply by wlandau** [reply]
Mentioning `tarchetypes` everywhere seems doable. I'd like to keep the current separation between packages because `targets` already does so much on its own.

**Reply by wlandau** [reply]
Just updated the help files and manual to mention `tarchetypes` alongside `targets`.
